File: Tree_Height_of_a_Binary_Tree.cpp

/*The tree node has data, left child and right child 
class Node {
    int data;
    Node* left;
    Node* right;
};

*/
int height(Node* root) {
	int heightL = 0;
	int heightR = 0;
	
	if(!root)
		return -1;
	else
	{
		heightL += height(root->left);
		heightR += height(root->right);
		
		//cout << "heightL:: " << heightL << " heightR:: " << heightR << endl;
	}
	
	return max(heightL, heightR) + 1;
}

File: Tree_Height_of_a_Binary_Tree.java


	/*
    class Node 
    	int data;
    	Node left;
    	Node right;
	*/
	public static int height(Node root) {
      	// Write your code here.
          int _height = 0;
          if(root != null)
          {
              if(root.left != null || root.right != null)
              {
                  _height++;
                  int leftHeight = height(root.left);
                  int rightHeight = height(root.right);
                  _height += Integer.max(leftHeight, rightHeight);
              }
          }
          return _height;
    }

